packages/ui/src/kitchenSinkItem.tsx

```diff
@@ -1,18 +1,23 @@
-import { h, ComponentChild, Ref } from "preact";
-import { getContribLogger } from "@playkit-js-contrib/common";
-import {
-    KitchenSinkExpandModes,
-    KitchenSinkItemData,
-    KitchenSinkPositions
-} from "./kitchenSinkItemData";
+import { ComponentChild, h } from "preact";
+import { ContribLogger, getContribLogger, Handler } from "@playkit-js-contrib/common";
+import { KitchenSinkItemData } from "./kitchenSinkItemData";
 import { ManagedComponent } from "./components/managed-component";
-import { ContribLogger } from "@playkit-js-contrib/common";
-import { KitchenSinkAdapter } from "./components/kitchen-sink-adapter";
+import { EventTypes, ItemActiveStateChangeEvent, ItemActiveStates } from "./kitchenSinkManager";
+import { KitchenSink } from "./components/kitchen-sink";
 
 export interface KitchenSinkItemOptions {
     data: KitchenSinkItemData;
-    isExpanded: (position: KitchenSinkPositions) => boolean;
-    expand: (position: KitchenSinkPositions, expandMode: KitchenSinkExpandModes) => void;
+    isActive: (item: KitchenSinkItem) => boolean;
+    activate: (item: KitchenSinkItem) => void;
+    deactivate: (item: KitchenSinkItem) => void;
+    onActivationStateChange: (
+        type: EventTypes,
+        handler: Handler<ItemActiveStateChangeEvent>
+    ) => void;
+    unregisterActivationStateChange: (
+        type: EventTypes,
+        handler: Handler<ItemActiveStateChangeEvent>
+    ) => void;
```

Comment:
I get what you tried here. but a more elegant way is to pass an eventManager owned by the manager (which I saw that you added in  the manager)
```
eventManager: EventManager<KitchenSinkEvents>()
```
and handle `listen` and `unlisten` relevant methods

Reply:
done